highway.tex

```diff
@@ -24,807 +26,1323 @@
+% Note that the confidence computed by the above definition is monotonic in the sense that whenever an additional equivocator $V$ is revealed in the bigger protocol state $\hat{\sigma}>\sigma$, one can reduce the previously found summit $C$ into $\hat{C}$ by removing units produced by $V$.
+% Then, $q$ will be possibly reduced by one, but it will be compensated by the $2\ww\big(E(\sigma)\big)$ factor in the equation.
 
-That means, to have a live blockchain we just need to make sure that eventually the honest validators' $n_v$ values become large enough for the lemma to apply. On the other hand, if the $n_v$ are very large, new blocks will only be created very infrequently and the blockchain's progress will be slow; so it's important to optimize the choice of these parameters.
+% The following refined version of the Lemma \ref{LemmaConfidence} describes the security guarantees obtained by the validators searching for honest-so-far summits:
 
-This is the motivation for the protocol's name: The nodes have different speed parameters $n_v$, as if they were driving on different lanes of a highway.
+% \begin{lemma}[\bf practical confidence]
+% Let $C = C_0\cup\dots\cup C_k$ be a $(q,k)$-summit for block $B$ in $\sigma$ consisting solely of honest-so-far validators, and $u$ be a unit such that $D(u)\cap C_k\neq\emptyset$.
+% Then either $\vote(u) \geq B$ or $\ww(\mathcal{V}_B) \geq \conf_{practical}(C)$.
+% \end{lemma}
 
 
-\subsection{Parameter strategy: Target Threshold for Finality}
 
-Given a family $\mathcal{F}$ of voting functions, a threshold $0\% < D < 100\%$, \emph{break parameters} $C_0 < C_1 \in \mathbb{N}$, and an \emph{acceleration parameter} $B \in \mathbb{N}$, we choose $n_v$ as follows. Given $i$, and $m = n_v(i - 1)$, whenever $2^m$ divides $i$ and $n_v$ was constant between $i - C_1 \cdot 2^m$ and $i - 1$:
-\begin{itemize}
-  \item If $i/2^m$ is even and fewer than $C_0$ out of all $k \in \{1, \ldots, C_1\}$ succeeded, then let $n_v(i) = m + 1$, where:\\
-    A $k \in \mathbb{N}$ \emph{succeeded} if there is an $f \in \mathcal{F}$ and a leader message $\lambda$ in tick $i - k \cdot 2^m$, such that $f$ was $\mathbb{0}$ for all messages in $J(\lambda)$, $f(\lambda) \neq \emptyset$, and $f(\lambda)$ was finalized with threshold $D$ in our local protocol state by time $i - (k - 1) 2^m$.
-  \item Otherwise, if $i/2^m$ is divisible by $B$, let $n_v(i) = m - 1$.
-\end{itemize}
-In other words, if fewer than $C_0$ out of $C_1$ consecutive leaders have successfully finalized a new value with threshold $D$, then we increase $n_v$; but every $B$ rounds, we optimistically decrease it. If both criteria apply, increasing $n_v$, i.e. slowing down, takes precedence.
+% \damian{Notation for validators/units $u,v$, $V$, $\mu, \tau$?}
```

Comment:
No, just bad habits. Overleaf is a very convenient tool when in comes to parallel work on a single document, one thing it's missing though is a solid version control system. As a consequence, if you use overleaf too much you start to think "well, I could just delete this line, but it may become handy later and then it's gonna be hard to restore. I'll just comment it out then".;)